pint_server/models.py

```diff
@@ -80,17 +80,26 @@ def validate_image_dates(self, key, value):
         deprecatedon = value if key == 'deprecatedon' else self.deprecatedon
         deletedon = value if key == 'deletedon' else self.deletedon
 
-        if deprecatedon and deprecatedon < publishedon:
-            raise ValueError('Image %s invalid dates specified - '
-                             'publishedon(%s) should not be after '
-                             'deprecatedon(%s)' % (self.name,
-                             str(publishedon), str(deprecatedon)))
-
-        if deletedon and deletedon < publishedon:
-            raise ValueError('Image %s invalid dates specified - '
-                             'publishedon(%s) should not be after '
-                             'deletedon(%s)' % (self.name,
-                             str(publishedon), str(deletedon)))
+        # If called for deprecatedon or deletedon before publishedon
```

Comment:
I think we still need to check to see if there's no garbage in.

```
if type(value) != str:
    raise ValueError("Expect the value of %s to be a string, got %s instead." % (key, type(value))
```

Reply:
Technically I believe that issue arose because the pytest setup in the pint-data-manager tests was using string date values rather than a valid datestamp type in it's mocking.

Also the above suggestion would actually break things because the values in question are not supposed to be strings at this point, but datestamps, as per the error case this PR is seeking to fix (`Failed to upgrade Pint database: '<' not supported between instances of 'datetime.date' and 'NoneType'`) so the suggested exception would in fact be triggered for valid values, rather than invalid ones.

We could instead explore leveraging the validator to coerce the provided value to be of the correct type, but I think https://docs.sqlalchemy.org/en/14/core/custom_types.html#typedecorator-recipes would be a better approach...